app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -16,6 +16,8 @@ def index
                             client_id: @authorize_form.client_id,
                             errors: @authorize_form.errors.messages)
 
+      return create if already_allowed?
```

Comment:
I thought about that, but it would involve calling `submit` again and grabbing the `:redirect_uri` key again from the result, so this just seemed easier? How strongly do you feel?